litestar/contrib/sqlalchemy/repository/async_repository.py

```diff
@@ -0,0 +1,560 @@
+        """
+        with wrap_sqlalchemy_exception():
+            item_id = self.get_id_attribute_value(data)
+            # this will raise for not found, and will put the item in the session
+            await self.get(item_id)
+            # this will merge the inbound data to the instance we just put in the session
+            instance = await self._attach_to_session(data, strategy="merge")
+            await self.session.flush()
+            self.session.expunge(instance)
+            return instance
+
+    async def update_many(self, data: list[ModelT]) -> list[ModelT]:
+        """Update one or more instances with the attribute values present on `data`.
+
+        This function has an optimized bulk insert based on the configured SQL dialect:
+        - For backends supporting `RETURNING` with `executemany`, a single bulk insert with returning clause is executed.
+        - For other backends, it does a bulk insert and then selects the inserted records
+
+        Args:
+            data: A list of instances to update.  Each should have a value for `self.id_attribute` that exists in the
+                collection.
+
+        Returns:
+            The updated instances.
+
+        Raises:
+            NotFoundError: If no instance found with same identifier as `data`.
+        """
+        data_to_update: list[dict[str, Any]] = [v.to_dict() if isinstance(v, self.model_type) else v for v in data]  # type: ignore
+        with wrap_sqlalchemy_exception():
+            if self._dialect.update_executemany_returning and self._dialect.name != "oracle":
+                instances = list(
+                    await self.session.scalars(
+                        update(self.model_type).returning(self.model_type),
+                        cast("_CoreSingleExecuteParams", data_to_update),  # this is not correct but the only way
+                        # currently to deal with an SQLAlchemy typing issue. See
+                        # https://github.com/sqlalchemy/sqlalchemy/discussions/9925
+                    )
+                )
+                await self.session.flush()
+                for instance in instances:
+                    self.session.expunge(instance)
+                return instances
+            await self.session.execute(
+                update(self.model_type),
+                data_to_update,
+            )
+            await self.session.flush()
+            return data
+
+    async def list_and_count(
+        self,
+        *filters: FilterTypes,
+        **kwargs: Any,
+    ) -> tuple[list[ModelT], int]:
+        """List records with total count.
+
+        Args:
+            *filters: Types for specific filtering operations.
+            **kwargs: Instance attribute value filters.
+
+        Returns:
+            Count of records returned by query, ignoring pagination.
+        """
+        if self._dialect.name in {"spanner", "spanner+spanner"}:
```

Comment:
@cofin is this correct? It used to be `in {"spanner"}`, on the async repo and `in {"spanner+spanner"}` on the sync repo, so I assumed this was the actual intention?